libs/shared/shared/reports/resources.py

```diff
@@ -270,7 +259,7 @@ def __contains__(self, filename):
         return filename in self._files
 
     @sentry_sdk.trace
-    def merge(self, new_report, joined=True):
+    def merge(self, new_report, joined=True, is_disjoint=False):
```

Comment:
I think a docstring explaining `is_disjoint` in this function would be nice, and also pointing out you need to call `finish_merge` at some point

(I know they are very close in the code)